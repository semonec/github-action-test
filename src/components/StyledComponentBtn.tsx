import { styled } from "styled-components";

// The Button from the last section without the interpolations
const Button = styled.button`
  color: #BF4F74;
  font-size: 1em;
  margin: 1em;
  padding: 0.25em 1em;
  border: 2px solid #BF4F74;
  border-radius: 3px;
`;

// A new component based on Button, but with some override styles
const TomatoButton = styled(Button)`
  color: tomato;
  border-color: tomato;
`;

const StyledComponentBtn = () => {
    return (
        <div>
            <Button>Normal Button </Button>
            <TomatoButton > Tomato Button </TomatoButton>
        </div>
    )
}

export default StyledComponentBtn