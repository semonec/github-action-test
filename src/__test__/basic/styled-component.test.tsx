import StyledComponentBtn from '../../components/StyledComponentBtn';

import React from 'react';
import renderer from 'react-test-renderer';

it('renders correctly', () => {
    const tree = renderer.create(
        <StyledComponentBtn />
    )
        .toJSON();
    expect(tree).toMatchSnapshot();
});