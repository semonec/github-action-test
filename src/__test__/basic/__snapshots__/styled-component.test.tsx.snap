// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div>
  <button
    className="sc-bcPKhP buQGRn"
  >
    Normal Button 
  </button>
  <button
    className="sc-bcPKhP sc-grXZZQ buQGRn blgqjQ"
  >
     Tomato Button 
  </button>
</div>
`;
